plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,97 @@
 #!/usr/bin/env python
 import re
 import sys
+import json
+import argparse
 from subprocess import CalledProcessError
+from collections import Counter
 
-from openshift import oc
+import openshift
 import nagios
 
 
 check_disk_cmd = ("df", "--output=target,pcent,ipcent")
 # Example output:
 # /etc/hosts      19%    8%
-check_disk_output_pattern = re.compile(r"([\w/]+)\s+(\d+)%\s+(\d+)%")
+check_disk_output_pattern = re.compile(
+    r"([A-Za-z0-9\./\-_]+)\s+(\d+)%\s+(\d+)%")
 
 
-def parse_df(line):
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--warn", type=int, action="store",
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
     return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
 
-def analize(pcent, ipcent, warning_threshold, critical_threshold):
-    max_pcent = max(pcent, ipcent)
-    return max_pcent >= warning_threshold, max_pcent >= critical_threshold
 
-def report():
-    raise NotImplementedError("report is not implemented yet")
+def parse_df_lines(lines):
+    return (map(parse_df_line, lines.splitlines()[1:]))
```

Comment:
It returns an empty tuple not really ignores it.  I thought it seemed cleaner to remove the line rather than have to post process the results of the map call to filter it out, but I can see how its probably better to avoid just blindly ignoring a line, so I've adjusted to use a list comprehension to remove empty tuple(s).